### PR TITLE
Added Member and Admin Dashboard

### PR DESCRIPTION
**Pull request by ThaiQ**
1. Made Member Dashboard (MDB) in Pages/Overview
2. Applying user API into MDB
3. Successfully implemented delete user from DB for MDB

### CODE REVIEW

File: test/frontend/MemberDashboard.test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/* global describe it */
+import 'jsdom-global/register'
+import React from 'react'
+import Enzyme, { mount } from 'enzyme'
+import { expect } from 'chai'
+
+import MemmberDashboard from '../../src/Pages/Overview/App.js'
```
Comment (evanugarte) [inline test/frontend/MemberDashboard.test.js]:
```suggestion
import MemberDashboard from '../../src/Pages/Overview/App.js'
```

File: test/frontend/MemberDashboard.test.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import Adapter from 'enzyme-adapter-react-16'
+import { Table } from 'reactstrap'
+
+Enzyme.configure({ adapter: new Adapter() })
+
+describe('<table />', () => {
```
Comment (evanugarte) [inline test/frontend/MemberDashboard.test.js]: I mentioned this before on #146. please keep the same [naming convention](https://github.com/SCE-Development/Core-v4/blob/7eb49330115efe0619c97377f7b69382cca2d41d/test/frontend/Slideshow.test.js#L13) for React tests.
```suggestion
describe('<MemberDashboard />', () => {
```

File: test/frontend/MemberDashboard.test.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  it('Should render a <table /> component with one child', () => {
+    const wrapper = mount(<MemmberDashboard />)
```
Comment (evanugarte) [inline test/frontend/MemberDashboard.test.js]:
```suggestion
    const wrapper = mount(<MemberDashboard />)
```

File: test/frontend/MemberDashboard.test.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import React from 'react'
+import Enzyme, { mount } from 'enzyme'
+import { expect } from 'chai'
+
+import MemberDashboard from '../../src/Pages/Overview/App.js'
+import Adapter from 'enzyme-adapter-react-16'
+
+Enzyme.configure({ adapter: new Adapter() })
+
+describe('<MemmberDashboard />', () => {
```
Comment (evanugarte) [inline test/frontend/MemberDashboard.test.js]:
```suggestion
describe('<MemberDashboard />', () => {
```

File: test/frontend/MemberDashboard.test.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    enumerable: true,
+    writable: true
+  })
+
+  it('Should render a <table /> component with one child', () => {
+    const wrapper = mount(<MemberDashboard />)
+    expect(wrapper.find('table')).to.have.lengthOf(1)
+  })
+
+  it('Should render a <table /> component with one child', () => {
```
Comment (evanugarte) [inline test/frontend/MemberDashboard.test.js]: Add another test case in here making sure that components will hold text values,

here is an [example](https://github.com/SCE-Development/Core-v4/blob/5965f2e1c00b49278abdf5a9a421f9025f1237d8/test/frontend/2dPrintingTest.js#L24)

File: src/Pages/MembershipPlan/App.css [hunk 6]
```diff
@@ -18,3 +18,56 @@
...
+
+  cursor:pointer;
+  /* background-color: rgb(78, 72, 72); */
+  padding: 5px;
+  border-radius: 20px;
+  outline: none;
+  /* border:none; */
+  fill:rgb(8, 185, 117);
+  animation: reverse; 
+}
```
Comment (evanugarte) [inline src/Pages/MembershipPlan/App.css]: Please add a newline to the end of the file

File: src/Pages/MembershipPlan/App.css [hunk 7]
```diff
@@ -18,3 +18,56 @@
   height: 200px;
   width: 100%;
 }
+.emailVerified{
```
Comment (evanugarte) [inline src/Pages/MembershipPlan/App.css]: for this file, please keep whitespace between css blocks, and remove any whitespace within css blocks

File: test/frontend/MemberDashboard.test.js (hunk 5 above)
Comment (ThaiQ) [reply test/frontend/MemberDashboard.test.js]: Can someone write this? I want to focus on getting the meat done

File: package.json [hunk 8]
```diff
@@ -59,6 +59,7 @@
     "express": "^4.16.2",
     "filepond": "^4.7.2",
     "isomorphic-fetch": "^2.2.1",
+    "jsdom": "^15.2.1",
```
Comment (evanugarte) [inline package.json]: revert this file back to what it was, we have jsdom

File: src/Pages/Profile/admin/AdminView.jsx [hunk 9]
```diff
@@ -0,0 +1,412 @@
...
+      user: {},
+      temp: {},
+      toggle: false,
+      toggleSubmit: false,
+      usernameCheck: false,
+    };
+  }
+
+  componentWillMount()
+  {
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: fix the brackets here.

File: src/Pages/Profile/admin/AdminView.jsx [hunk 10]
```diff
@@ -0,0 +1,412 @@
...
+    this.setState({user:{...this.props.user}},
+      this.setState({temp:{...this.props.user}})
+    )
+  }
+
+  handleToggle()
+  {
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: fix the brackets here.

File: src/Pages/Profile/admin/AdminView.jsx [hunk 11]
```diff
@@ -0,0 +1,412 @@
...
+    this.setState({temp:{...this.props.user}},
+      this.setState({toggle: !this.state.toggle})
+    )
+  }
+
+  handleSubmission()
+  {
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: fix the brackets here.

File: src/Pages/Profile/admin/AdminView.jsx [hunk 12]
```diff
@@ -0,0 +1,412 @@
...
+    const queryEmail = this.state.user.email
+
+    this.setState({user:{...this.state.temp}},
+      this.setState({toggle:!this.state.toggle},
+        ()=>
+        {
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: fix the brackets here.

File: src/Pages/Profile/admin/AdminView.jsx [hunk 13]
```diff
@@ -0,0 +1,412 @@
...
+              console.log(err)
+            })
+            this.handleSubmissionToggle()
+        }
+      )
+    )
+  }
+
+  handleSubmissionToggle()
+  {
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: fix the brackets here and the rest below.

File: src/Pages/Profile/admin/AdminView.jsx [hunk 14]
```diff
@@ -0,0 +1,412 @@
...
+    if (this.state.user.email===this.state.temp.email)
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: add spacing between `===`

File: test/frontend/MemberDashboard.test.js (hunk 5 above)
Comment (evanugarte) [reply test/frontend/MemberDashboard.test.js]: your component, your tests man

File: src/Pages/Profile/admin/AdminView.jsx [hunk 15]
```diff
@@ -0,0 +1,412 @@
+import React, { Component } from "react";
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: This file is much to complicated. Try breaking it into components, or consolidate the if statements

File: src/Pages/Profile/admin/AdminView.jsx [hunk 16]
```diff
@@ -0,0 +1,355 @@
...
+  /*
+  translating access Level into role
+  The access level is defined as follows:
+  -2: Ban
+  -1: Pending
+   0: Member
+   1: Officer
+   2: Admin
+  */
+  function roleTranslator(level) {
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: lets add the enums in this PR. this is duplicated twice across Overview and AdminView

File: src/Pages/Profile/admin/AdminView.jsx [hunk 17]
```diff
@@ -0,0 +1,355 @@
...
+    /*
+    translating access Level into role
+    The access level is defined as follows:
+    -2: Ban
+    -1: Pending
+     0: Member
+     1: Officer
+     2: Admin
+    */
+    let memberships = [0, 1, -1, 2, -2]
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: this will become enums

File: src/Pages/Profile/admin/AdminView.jsx [hunk 18]
```diff
@@ -0,0 +1,355 @@
...
+                  <Input
+                    type={group.type}
+                    name={group.type}
+                    placeholder={group.placeholder}
+                    onChange={group.onChange}
+                  />
+                </FormGroup>
+              })}
+
+              {/*Need Improvements, not doing anything currently*/}
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: Remove the commented out code below. you can open an issue to add photo upload and paste the code there

File: src/Pages/Profile/admin/AdminView.jsx [hunk 19]
```diff
@@ -0,0 +1,355 @@
...
+              Cancel
+            </Button>
+          </ModalFooter>
+
+        </Modal>
+      </div>
+    );
+  }
+
+  function display() {
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: this should be it's own component. each thing to be rendered should be sent as props.

File: src/Pages/Profile/admin/AdminView.jsx [hunk 20]
```diff
@@ -0,0 +1,355 @@
...
+            <Modal
+              style={
+                {
+                  marginTop: '320px',
+                }}
+              isOpen={toggleSubmit}>
+              <Button
+                onClick={async () => { await handleSubmission() }}
+                color="primary">
+                YES! mutate them!
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]:
```suggestion
                Yes
```

File: src/Pages/Profile/admin/AdminView.jsx [hunk 21]
```diff
@@ -0,0 +1,355 @@
...
+    The access level is defined as follows:
+    -2: Ban
+    -1: Pending
+     0: Member
+     1: Officer
+     2: Admin
+    */
+    let memberships = [0, 1, -1, 2, -2]
+
+    return (
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: the code below this line is super confusing. there are three `return` statements. please split them into individual components

File: src/Pages/Overview/App.js [hunk 22]
```diff
@@ -0,0 +1,343 @@
...
+        return 'Officer'
+      case 2:
+        return 'Admin'
+      default:
+        return 'Pending'
+    }
+  }
+
+  // displaying children of board
+  display (user, index) {
```
Comment (evanugarte) [inline src/Pages/Overview/App.js]: this should be it's own component. see if you can make a component that you can use across overview and admin.

File: src/Pages/Overview/App.js [hunk 23]
```diff
@@ -0,0 +1,343 @@
...
+  /*
+  translating access Level into role
+  The access level is defined as follows:
+  -2: Ban
+  -1: Pending
+   0: Member
+   1: Officer
+   2: Admin
+  */
+  roleTranslator (level) {
```
Comment (evanugarte) [inline src/Pages/Overview/App.js]: as mentioned below, enums!

File: src/Pages/Overview/App.js [hunk 24]
```diff
@@ -0,0 +1,343 @@
...
+    const search = this.state.search.trim().toLowerCase()
+    return search !== null || search !== ''
+      ? this.state.users.filter(data =>
+        data.firstName.toLowerCase().includes(search.toLowerCase())
+      )
+      : this.state.users
+  }
+
+  // Return check mark if true
+  marks (sign) {
```
Comment (evanugarte) [inline src/Pages/Overview/App.js]: this should be a component

File: src/Pages/Overview/App.js [hunk 25]
```diff
@@ -0,0 +1,343 @@
...
+Ya are my ace good luck.
+*/
+
+import React, { Component } from 'react'
+import './App.css'
+import InfoCard from '../Profile/admin/AdminView'
+import { Modal, ModalHeader, ModalBody, ModalFooter, Button } from 'reactstrap'
+
+import axios from 'axios'
+class Login extends Component {
```
Comment (evanugarte) [inline src/Pages/Overview/App.js]: Take a look at how we used hooks in the other file and use them here. Also this component should't be named login.

File: src/Enums.js [hunk 26]
```diff
@@ -0,0 +1,31 @@
+export const membershipStatus = [
```
Comment (evanugarte) [inline src/Enums.js]: you can also do
```js
export const membershipStatus = {
  BAN: -2,
  PENDING: -1,
   // ...
}
```

that is likely better than string to int mapping as you can use the `membershipStatus` variables directly

File: src/Pages/Overview/App.js [hunk 27]
```diff
@@ -0,0 +1,217 @@
...
+    this.setState({ search: event.target.value })
+  }
+
+  /*
+  Delete api
+  parameter: Json object of object to be deleted
+  */
+  deleteUser (user) {
+    // if they decide to delete themselves: delete->logout
+    if (user.email === this.state.currentUser) {
```
Comment (evanugarte) [inline src/Pages/Overview/App.js]: the axios call in the if and else statement is exaclty the same. instead, move the call ouside of the if/else block, and just keep the logout stuff in a single if statement

File: src/Pages/Overview/App.js [hunk 28]
```diff
@@ -0,0 +1,217 @@
...
+        <input
+          onChange={e => {
+            this.updateSearch(e)
+          }}
+        />
+
+        <table className='content-table' id='users'>
+          <thead>
+            <tr>
+              <th>Name</th>
```
Comment (evanugarte) [inline src/Pages/Overview/App.js]: use array.map here
```jsx
// ['Name', 'Door Code', 'Printing', ...]
this.state.userColumns.map((title) => {
  return (<th>{title}</th>)
});
```

File: src/Pages/Overview/App.js [hunk 29]
```diff
@@ -0,0 +1,217 @@
...
+              No, they're chill
+            </Button>
+          </ModalFooter>
+        </Modal>
+      </div>
+    )
+  }
+}
+
+export default Login
```
Comment (evanugarte) [inline src/Pages/Overview/App.js]: as mentioned before this component shouldn't be named Login. Also rename the filename from App.js. see #170

File: src/Pages/Overview/DisplayProfile.js [hunk 30]
```diff
@@ -0,0 +1,101 @@
+import React from 'react'
```
Comment (evanugarte) [inline src/Pages/Overview/DisplayProfile.js]: I understand that the intention here is to try and simplify things, but this splits functions between two files making things confusing. Instead,

1. make `displayBoard` its own react component. (send parameters via props)
1. move all non html rendering functions back in their respective classes
1. take the `marks` function and make it its own react component. (send parameters via props)

File: src/Pages/Overview/DisplayProfile.js [hunk 31]
```diff
@@ -0,0 +1,101 @@
...
+          user.lastName.slice(1, user.lastName.length)}
+        {user.middleInitial.trim() !== '' &&
+          ' ' + user.middleInitial.toUpperCase() + '.'}
+      </td>
+
+      <td>{user.doorCode}</td>
+
+      <td>{user.pagesPrinted}/30</td>
+
+      <td>{this.marks(user.emailVerified)}</td>
```
Comment (evanugarte) [inline src/Pages/Overview/DisplayProfile.js]: as stated at the top of this file, make marks its own react component

File: src/Pages/Overview/DisplayProfile.js [hunk 32]
```diff
@@ -0,0 +1,101 @@
...
+import './App.css'
+const enums = require('../../Enums.js')
+
+let toggle = false
+let toggleDelete = false
+let userToDelete = null
+let userToEdit = null
+
+// displaying children of member-board
+export function displayBoard (user, index) {
```
Comment (evanugarte) [inline src/Pages/Overview/DisplayProfile.js]: Make `displayboard` it's own react component, and give it a better name for example `<OverviewProfile />`

File: src/Pages/Overview/DisplayProfile.js [hunk 33]
```diff
@@ -0,0 +1,101 @@
...
+  return (
+    <tr key={index}>
```
Comment (evanugarte) [inline src/Pages/Overview/DisplayProfile.js]: You can remove the `key` element here. This component is not direclty inside of an `array.map` so it is ok. See [this article](https://reactjs.org/docs/lists-and-keys.html#keys) for more details

File: src/Pages/Overview/DisplayProfile.js (hunk 32 above)
Comment (evanugarte) [inline src/Pages/Overview/DisplayProfile.js]: Remove `index` parameter

File: src/Pages/Overview/DisplayProfile.js [hunk 34]
```diff
@@ -0,0 +1,101 @@
...
+      </td>
+
+      <button
+        className='delete'
+        onClick={() => {
+          userToEdit = user
+          toggler()
+        }}
+      >
+        <svg width='40' height='40' viewBox='0 0 24 24'>
```
Comment (evanugarte) [inline src/Pages/Overview/DisplayProfile.js]: I have no idea what this SVG is supposed to show when i look at the code. what would be good is to put each SVG in it's own component, for example `<GreenCheck />` is much easier to understand and import when needed

File: src/Pages/Overview/DisplayProfile.js [hunk 35]
```diff
@@ -0,0 +1,101 @@
...
+         14.11C13.34,14.03 12.67,14 12,14M12,4A4,4 0 0,0 8,8A4,4 0 0,0 12,12A4,4 0 0,0 16,8A4,4 0 0,0 12,4Z'
+          />
+        </svg>
+      </button>
+    </tr>
+  )
+}
+
+// Return check mark if true
+export function marks (sign) {
```
Comment (evanugarte) [inline src/Pages/Overview/DisplayProfile.js]: Make marks its own component as stated before

File: src/Pages/Overview/DisplayProfile.js [hunk 36]
```diff
@@ -0,0 +1,101 @@
...
+      <path d='M20.285 2l-11.285 11.567-5.286-5.011-3.714 3.716 9 8.728 15-15.285z' />
+    </svg>
+  ) : (
+    <svg width='24' height='24' viewBox='0 0 24 24' style={{ fill: 'RED' }}>
+      <path d='M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z' />
+    </svg>
+  )
+}
+
+export function getToggle () {
```
Comment (evanugarte) [inline src/Pages/Overview/DisplayProfile.js]: `Lines 79-end` can be moved back in their respective components

File: src/Pages/Profile/admin/AdminView.jsx [hunk 37]
```diff
@@ -0,0 +1,294 @@
...
+        }
+      })
+      .catch(err => {
+        window.alert('The Email has already existed, leave it blank if you do not want to change')
+        console.log(err)
+      })
+  }
+
+  function editModalButton() {
+    let FormGroups = [
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: remove the `type` field for each object, except for when there is a password. see comment on `Line 179` for fixing the map function

Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]:
```suggestion
    const formGroups = [
```

File: src/Pages/Profile/admin/AdminView.jsx [hunk 38]
```diff
@@ -0,0 +1,294 @@
...
+            Edit
+          </Button>
+        </div>
+
+        <Modal isOpen={toggle}>
+          <ModalHeader>Modal title</ModalHeader>
+          <ModalBody>
+            <Form>
+
+              {FormGroups.map((group, index) => {
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]:
```suggestion
              {formGroups.map((group, index) => {
```

File: src/Pages/Profile/admin/AdminView.jsx [hunk 39]
```diff
@@ -0,0 +1,294 @@
...
+                return <FormGroup key={index}>
+                  <Label>{group.label}</Label>
+                  <Input
+                    type={group.type}
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]:
```suggestion
                    type={group.type || 'email'}
```

File: src/Pages/Profile/admin/AdminView.jsx [hunk 40]
```diff
@@ -0,0 +1,294 @@
...
+                    name={group.type}
+                    placeholder={group.placeholder}
+                    onChange={group.onChange}
+                  />
+                </FormGroup>
+              })}
+
+              Change validation date to
+              <select onChange={(e)=>{setNumberOfSemestersToSignUpFor(e.target.value)}}>
+                <option value={0}>Keep Same</option>
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: use array.map here. you can use index for the `<option value />`

File: src/Pages/Profile/admin/AdminView.jsx [hunk 41]
```diff
@@ -0,0 +1,294 @@
...
+                  setPagesPrinted(0);
+                }}
+                color="info"
+                style={{marginTop:'5px'}}>
+                Reset Pages!
+              </Button>
+
+              <FormGroup tag="fieldset">
+                <legend>Membership Status</legend>
+                {enums.getAllValues(enums.membershipStatus).map((membership, index) => {
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: This `return` block should be it's own react component. i dont wanna merge things that may be hard to maintain later. see #61

File: src/Pages/Profile/admin/AdminView.jsx [hunk 42]
```diff
@@ -0,0 +1,294 @@
...
+              Cancel
+            </Button>
+          </ModalFooter>
+
+        </Modal>
+      </div>
+    );
+  }
+
+  return (
```
Comment (evanugarte) [inline src/Pages/Profile/admin/AdminView.jsx]: this return block should also be its own react component.

i understand that refactoring this is cumbersome, but we are going for maintainability. last thing i'd want is all our hard work to get deprecated.

File: src/Pages/Profile/admin/profile.js [hunk 43]
```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import { Badge } from 'reactstrap'
+const enums = require('../../../Enums.js')
+
+export function displayProfile (user, membershipValidUntil) {
```
Comment (evanugarte) [inline src/Pages/Profile/admin/profile.js]:
```suggestion
export function Profile (props) {
```
use props here. it's the convention for [functional components](https://programmingwithmosh.com/react/react-functional-components/)

File: test/frontend/MemberDashboard.test.js (hunk 4 above)
Comment (evanugarte) [reply test/frontend/MemberDashboard.test.js]: This is not resolved. please rename the test case like above.

File: src/Enums.js (hunk 26 above)
Comment (ThaiQ) [reply src/Enums.js]: the problem with that is idk how to get the key from value

File: src/Pages/Overview/DisplayProfile.js (hunk 33 above)
Comment (ThaiQ) [reply src/Pages/Overview/DisplayProfile.js]: Console give a key-warning when remove. Having it there solve the warning.

File: src/Pages/Overview/DisplayProfile.js (hunk 32 above)
Comment (ThaiQ) [reply src/Pages/Overview/DisplayProfile.js]: i will use hook

File: src/Pages/Overview/DisplayProfile.js (hunk 30 above)
Comment (ThaiQ) [reply src/Pages/Overview/DisplayProfile.js]: I will move all the SVG into a js file and return functions instead. I think this is more clean

**Pull request closed without merging**